frontend/www/js/omegaup/components/problem/Details.vue

```diff
@@ -466,6 +467,15 @@ export default class ProblemDetails extends Vue {
     return this.runs.filter((run) => run.alias === this.problem.alias);
   }
 
+  get filteredLanguages(): string[] {
+    if (!this.languages) {
+      return this.problem.languages;
+    }
+    return this.problem.languages.filter((language) =>
+      this.languages?.includes(language),
+    );
```

Comment:
```suggestion
    const languagesSet = new Set(this.languages);
    return this.problem.languages.filter((language) =>
      languagesSet.has(language),
    );
```

siempre hay que evitar comportamiento potencialmente cuadrático usando [`Set`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set).

Reply:
Listo.